### ISSUE TITLE
Feature Request: Configurable delay for Amplipi output

### ISSUE THREAD

**Issue by iloveicedgreentea** [micro-nova/AmpliPi#458]
TLDR: given an input signal, I want to add delay per zone/group on the amplipi

Using a setup where I use an AVR to power one zone and amplipi for the rest, the audio is not synced. I need to delay the amplipi by 50-200ms to get it synced up.

Happy to work on this if you can point me in the right direction as to where additions would go.

The specific use case:

1. User is using an AVR for zone 1
2. User is using amplipi as airplay source (other sources may have same problem)
3. Amplipi is sending zone 1 preout to AVR input

The problem:

1. Zone 1 has perceptible delay because the AVR adds latency to the stream
2. Amplipi cannot add delay to other zones

So there would need a solution which adds "negative delay" - Amplipi would need to have a way to add delay by zone so that zone 2-12 can be delayed to sync up with zone 1.

**Comment by linknum23** [comment]
The underlying airplay stream (shairport-sync) has support for a configurable delay but the rest of the streams supported by AmpliPi do not so we haven't exposed that configuration yet. 

In the future we could expose the delay parameter to the api and the ui. I think that would solve your use case.

In the meantime you have a couple of options:
- modify amplipi's shairport-sync config file generation in /home/pi/amplipi-dev/amplipi/streams.py to add the delay needed
- Route a preamp RCA output from your AVR into one of your AmpliPi's RCA inputs and use that to play to the rest of the zones

**Comment by iloveicedgreentea** [comment]
Thanks I will look into that and see if I can help expose that to the UI. Do I need to restart amplipi after changing this?

**Comment by iloveicedgreentea** [comment]
I also noticed in the file config code, its using audio_backend_buffer_desired_length under alsa, but in the shairplay docs this is actually listed under general

https://github.com/micro-nova/AmpliPi/blob/main/amplipi/streams.py#L268


https://github.com/mikebrady/shairport-sync/blob/master/scripts/shairport-sync.conf

Seems wrong, unless I am missing something?

**Comment by iloveicedgreentea** [comment]
While I think exposing this setting might solve another problem (like lipsync) I don't think it will fix my issue. The airplay stream is going into the amplipi, then one leg is going to the AVR. What I need is a way to delay only the signal in amplipi

If I add _audio_backend_latency_offset_in_seconds_ then it will have the same result, with the AVR signal being delayed.

**Comment by linknum23** [comment]
I see that now. I think we misunderstood your setup. 

For this setup the current hardware does not have the capability to correct for delays. This is something we will consider in the future designs.

**Comment by iloveicedgreentea** [comment]
Theres no way to even do it at a system level with pulse or something? I don't care about per stream/source.

**Comment by linknum23** [comment]
The current system does not have the capability to do any additional per zone adjustments. We are only able to make audio adjustments per stream.

It might help to look at AmpliPi's system diagram:

![image](https://github.com/micro-nova/AmpliPi/assets/628667/689d496c-0dd6-4042-b141-39374857ddf0)

The pi is able to output 4 stereo streaming outputs (source 1-4). Each zone only has control of which source to use and its volume.
